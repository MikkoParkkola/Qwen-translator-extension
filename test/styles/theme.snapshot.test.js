const fs = require('fs');
const path = require('path');

describe('apple theme snapshots', () => {
  const vars = [
    '--qwen-bg',
    '--qwen-text',
    '--qwen-border',
    '--qwen-error',
    '--qwen-input-bg',
    '--qwen-input-focus',
    '--qwen-primary-bg',
    '--qwen-primary-hover',
    '--qwen-secondary-bg',
    '--qwen-secondary-hover',
  ];

  beforeAll(() => {
    const css = fs.readFileSync(path.resolve(__dirname, '../../src/styles/apple.css'), 'utf8');
    const style = document.createElement('style');
    style.textContent = css;
    document.head.appendChild(style);
  });

  const snapshotFor = el => {
    const computed = getComputedStyle(el);
    const out = {};
    for (const v of vars) {
      out[v] = computed.getPropertyValue(v).trim();
    }
    return out;
  };

  test('dark mode', () => {
    const el = document.createElement('div');
    el.setAttribute('data-qwen-theme', 'apple');
    document.body.appendChild(el);
    expect(snapshotFor(el)).toMatchSnapshot();
  });

  test('light mode', () => {
    const el = document.createElement('div');
    el.setAttribute('data-qwen-theme', 'apple');
    el.setAttribute('data-qwen-color', 'light');
    document.body.appendChild(el);
    expect(snapshotFor(el)).toMatchSnapshot();
  });
});
