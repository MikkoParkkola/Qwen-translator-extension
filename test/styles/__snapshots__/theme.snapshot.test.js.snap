// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`apple theme snapshots dark mode 1`] = `
{
  "--qwen-bg": "rgba(28, 28, 30, 0.7)",
  "--qwen-border": "rgba(255, 255, 255, 0.2)",
  "--qwen-error": "#ff3b30",
  "--qwen-input-bg": "rgba(255, 255, 255, 0.1)",
  "--qwen-input-focus": "#0a84ff",
  "--qwen-primary-bg": "#0a84ff",
  "--qwen-primary-hover": "#0060df",
  "--qwen-secondary-bg": "rgba(118, 118, 128, 0.2)",
  "--qwen-secondary-hover": "rgba(118, 118, 128, 0.3)",
  "--qwen-text": "#f5f5f7",
}
`;

exports[`apple theme snapshots light mode 1`] = `
{
  "--qwen-bg": "rgba(255, 255, 255, 0.75)",
  "--qwen-border": "rgba(0, 0, 0, 0.1)",
  "--qwen-error": "#ff3b30",
  "--qwen-input-bg": "rgba(255, 255, 255, 0.9)",
  "--qwen-input-focus": "#0a84ff",
  "--qwen-primary-bg": "#0a84ff",
  "--qwen-primary-hover": "#0060df",
  "--qwen-secondary-bg": "rgba(118, 118, 128, 0.12)",
  "--qwen-secondary-hover": "rgba(118, 118, 128, 0.2)",
  "--qwen-text": "#1c1c1e",
}
`;
